src/manifest.c

```diff
@@ -428,13 +430,25 @@ int match_manifests(struct manifest *m1, struct manifest *m2)
 			file3->is_config = file1->is_config;
 			file3->is_state = file1->is_state;
 			file3->is_boot = file1->is_boot;
+			/* ghost deleted boot files */
+			file3->is_ghosted = file1->is_boot;
```

Comment:
Should this be `= file1->is_boot && file1->is_deleted`?

Reply:
If you look up a few lines you'll see that file1 is deleted if we get to this point, but it is nice to add the extra check here in case our logic fails (although that would never ever happen of course /s).